### PR TITLE
[#1760] Remove redundant methods in abstractHttpEndpoint

### PR DESCRIPTION
**Pull request by jbtrystram**
Also deprecate methods tied to the old event bus implementation. See #1760 

Signed-off-by: Jean-Baptiste Trystram <jbtrystram@redhat.com>

### CODE REVIEW

**Comment by sophokles73** [comment]
@jbtrystram can you rebase? I have moved the old event bus related functionality to their own (deprecated) subclass ...

**Comment by dejanb** [comment]
@jbtrystram  Looks good so far. I would also move static methods from `Util` class we added recently, to the `AbstractHttpEndpoint`

File: service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java [hunk 1]
```diff
@@ -322,4 +294,146 @@ protected final CorsHandler createCorsHandler(final String allowedOrigin, final
...
+     * Creates a new <em>OpenTracing</em> span for tracing the execution of a service operation.
+     *
+     * @param operationName The operation name that the span should be created for.
+     * @param spanContext Existing span context.
+     * @param tracer the Tracer instance.
+     * @param className The class name to insert in the Span.
+     * @return The new {@code Span}.
+     * @throws NullPointerException if operationName is {@code null}.
+     */
+    public static final Span newChildSpan(final String operationName, final SpanContext spanContext,
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java]: I would leave span related methods in the separate class, as they are used outside of these endpoints.

File: service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java [hunk 2]
```diff
@@ -322,4 +294,146 @@ protected final CorsHandler createCorsHandler(final String allowedOrigin, final
...
+     * <li>Ends a response.</li>
+     * </ol>
+     *
+     * @param ctx The routing context of the request.
+     * @param result The generic result of the operation.
+     * @param customHandler An (optional) handler for post processing successful HTTP response, e.g. to set any additional HTTP
+     *                      headers. The handler <em>must not</em> write to response body. May be {@code null}.
+     * @param span The active OpenTracing span for this operation. The status of the response is logged and span is finished.
+     */
+    public static final void writeResponse(final RoutingContext ctx, final Result<?> result, final Handler<HttpServerResponse> customHandler, final Span span) {
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java]: These methods should stay here, but as regular protected methods.

**Pull request merged**